File: docs-examples/example-function-java/src/test/java/example/ListBucketsFunctionTest.java
package example;

import com.fnproject.fn.testing.FnTestingRule;
import example.mock.MockData;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ListBucketsFunctionTest {

    FnTestingRule fn = FnTestingRule.createDefault();

    @Test
    void testFunction() {

        MockData.bucketNames.clear();
        MockData.bucketNames.add("b1");
        MockData.bucketNames.add("b2");

        fn.givenEvent().enqueue()
            .addSharedClass(MockData.class)
            .thenRun(ListBucketsFunction.class, "handleRequest");

        String body = fn.getOnlyResult().getBodyAsString();
        assertEquals("[\"b1\",\"b2\"]", body);
    }
}

File: docs-examples/example-function-kotlin/src/test/kotlin/example/ListBucketsFunctionTest.kt
package example

import com.fnproject.fn.testing.FnTestingRule
import example.mock.MockData
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class ListBucketsFunctionTest {

    var fn: FnTestingRule = FnTestingRule.createDefault()

    @Test
    fun testFunction() {
        MockData.bucketNames.clear()
        MockData.bucketNames.add("b1")
        MockData.bucketNames.add("b2")

        fn.givenEvent().enqueue()
            .addSharedClass(MockData::class.java)
            .thenRun(ListBucketsFunction::class.java, "handleRequest")

        val body = fn.onlyResult.bodyAsString
        assertEquals("[\"b1\",\"b2\"]", body)
    }
}
